ci/docker_cache.py

```diff
@@ -28,237 +28,164 @@
-    remote_image_id = _get_remote_image_id(s3_object)
-    if remote_image_id:
-        if _docker_layer_exists(remote_image_id):
-            logging.info('Local docker cache already present for {}'.format(docker_tag))
-            return
-        else:
-            logging.info('Local docker cache not present for {}'.format(docker_tag))
-
-        # Download using public S3 endpoint (without requiring credentials)
-        with tempfile.TemporaryDirectory() as temp_dir:
-            tar_file_path = os.path.join(temp_dir, 'layers.tar')
-            s3_object.download_file(
-                Filename=tar_file_path,
-                Callback=ProgressPercentage(object_name=docker_tag, size=s3_object.content_length))
-
-            # Load layers
-            cmd = ['docker', 'load', '-i', tar_file_path]
-            try:
-                check_call(cmd)
-                logging.info('Docker cache for {} loaded successfully'.format(docker_tag))
-            except CalledProcessError as e:
-                logging.error('Error during load of docker cache for {} at {}'.format(docker_tag, tar_file_path))
-                logging.exception(e)
-                return
-    else:
-        logging.info('No cached remote image of {} present'.format(docker_tag))
-
-
-def _docker_layer_exists(layer_id) -> bool:
-    """
-    Check if the docker cache contains the layer with the passed id
-    :param layer_id: layer id
-    :return: True if exists, False otherwise
-    """
-    cmd = ['docker', 'images', '-q']
-    image_ids_b = subprocess.check_output(cmd)
-    image_ids_str = image_ids_b.decode('utf-8').strip()
-    return layer_id in [id.strip() for id in image_ids_str.split('\n')]
+    # We don't have to retag the image since it's already in the right format
+    logging.info('Loading Docker cache for %s from %s', docker_tag, registry)
+    pull_cmd = ['docker', 'pull', docker_tag]
+    subprocess.call(pull_cmd)  # Don't throw an error if the image does not exist
 
 
-def _get_aws_session() -> boto3.Session:  # pragma: no cover
+def delete_local_docker_cache(docker_tag):
     """
-    Get the boto3 AWS session
-    :return: Session object
+    Delete the local docker cache for the entire docker image chain
+    :param docker_tag: Docker tag
+    :return: None
     """
-    global cached_aws_session
-    if cached_aws_session:
-        return cached_aws_session
-
-    session = boto3.Session()  # Uses IAM user credentials
-    cached_aws_session = session
-    return session
+    history_cmd = ['docker', 'history', '-q', docker_tag]
 
+    try:
+        image_ids_b = subprocess.check_output(history_cmd)
+        image_ids_str = image_ids_b.decode('utf-8').strip()
+        layer_ids = [id.strip() for id in image_ids_str.split('\n') if id != '<missing>']
 
-def _format_docker_cache_filepath(output_dir, docker_tag) -> str:
-    return os.path.join(output_dir, docker_tag.replace('/', '_') + '.tar')
+        delete_cmd = ['docker', 'image', 'rm', '--force']
```

Comment:
deleting in a script looks scary... are there alternatives?

Reply:
Could you elaborate?

Reply:
That's simply what you have to do if you want to clean the local docker cache